elk/training/reporter.py

```diff
@@ -268,9 +269,15 @@ def fit(
         best_state: dict[str, torch.Tensor] = {}  # State dict of the best run
         x0, x1 = contrast_pair
 
-        for _ in range(cfg.num_tries):
+        for i in range(cfg.num_tries):
             self.reset_parameters()
 
+            # This is sort of inefficient but whatever
```

Comment:
So you put the PCA init outside the reset_parameters() call because it needs access to the index i? IIUK, a new probe is trained for each of the top `num_tries` PCs. In future PRs it might be good to clean this up a bit and add explanation (and like you mentioned not recomputing the top i principal components every time).